MetaMask/metamask-mobile
# Update controller packages to match core v40

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

The controller packages have been updated to match the versions in the core monorepo v40 release. The keyring controller update was held back due to incompatibilities related to BigInt.

The only breaking change was to the network controller state. The state property `properties` was renamed to `networkDetails`. Luckily there was only a single direct reference to this property (in a test) so the number of changes required was minimal, but we did need a state migration.

Release notes: https://github.com/MetaMask/core/releases/tag/v40.0.0

**Manual Testing Instructions**

This release includes a bug fix related to NFTs, details here: https://github.com/MetaMask/core/pull/1082

We might be able to test that the issue is resolved by doing something like this:
* Setup a test account with an NFT
* Send the NFT away, and check to ensure that it is recognized as "previously owned" (I am unfamiliar with this feature so I'm not sure about this step)
* Send the NFT back to the test account
* Ensure that the NFT is recognized as owned, not previously owned

I am not confident in those steps though.

For the network controller update, there should be no functional changes.  There is a state migration, but it's hard to see the effects of that migration as a user (effectively it saves a single network call upon app load). The state gets overwritten upon each network switch. The e2e tests should suffice to catch unexpected network related regressions.

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/798

**Checklist**
